internal/xds/translator/listener.go

```diff
@@ -44,12 +45,24 @@ func http1ProtocolOptions(opts *ir.HTTP1Settings) *corev3.Http1ProtocolOptions {
 	if opts == nil {
 		return nil
 	}
-	if opts.EnableTrailers {
-		return &corev3.Http1ProtocolOptions{
-			EnableTrailers: opts.EnableTrailers,
+	if !opts.EnableTrailers && !opts.PreserveHeaderCase {
+		return nil
+	}
+	r := &corev3.Http1ProtocolOptions{
```

Comment:
trailers are enabled by default ?

Reply:
No, they are enabled only if `opts.EnableTrailers == true`.

This function returns `nil` only if both `EnableTrailers` and `PreserveHeaderCase` are false. If `PreserveHeaderCase` is true and `EnableTrailers` is false then setting the `EnableTrailers` field to false is simply keeping it at its default value of "disabled".

Reply:
yah thanks for explaining that, that logic should work

Reply:
can you address these as comment in the code?

Reply:
:-) 
The PR was merged before I could add this as a comment.

Reply:
you can open another one

Reply:
Added the comment in #2517